eventsource-client/examples/tail.rs

```diff
@@ -30,15 +30,16 @@ async fn main() -> Result<(), es::Error> {
         )
         .build();
 
-    let mut stream = Box::pin(tail_events(client));
+    let stream = client.stream();
+    let mut stream = tail_events(client.stream());
+
     while let Ok(Some(_)) = stream.try_next().await {}
 
     Ok(())
 }
 
-fn tail_events(client: es::Client<es::HttpsConnector>) -> impl Stream<Item = Result<(), ()>> {
-    client
-        .stream()
+fn tail_events(stream: BoxStream<es::Result<Event>>) -> impl Stream<Item = Result<(), ()>> + '_ {
+    stream
```

Comment:
This works, but I'm not sure why. I don't understand why I need to include this lifetime.